desc/grid.py

```diff
@@ -187,17 +187,25 @@ def _scale_weights(self):
     @property
     def L(self):
         """int: Radial grid resolution."""
-        return self.__dict__.setdefault("_L", 0)
+        if getattr(self, "_L", None) is None:
+            self._L = self.num_rho - 1
+        return self._L
 
     @property
     def M(self):
         """int: Poloidal grid resolution."""
-        return self.__dict__.setdefault("_M", 0)
+        if getattr(self, "_M", None) is None:
+            self._M = (
+                (self.num_theta // 2 - 1) if self.sym else (self.num_theta - 1) // 2
```

Comment:
I think these are reversed? If sym is true then there are fewer nodes for a given M

Reply:
The logic in create nodes is the opposite. Due to grid spacing stuff iirc
```python
theta = 2 * (M + 1) if self.sym else 2 * M + 1
```

Reply:
With this way rn,

```python
g = LinearGrid(theta=3,rho=1,sym=True)
```
results in `g.M` being `0`, which feels odd since it does have some sort of theta resolution, it is not just a single point.